agents/unix/conf/base/conf_sensor.c

```diff
@@ -0,0 +1,552 @@
+        if (subfeature != NULL)
+            break;
+    }
+    if (subfeature != NULL)
+    {
+        err = sensors_get_value(chip, subfeature->number,
+                                &subfeature_value);
+        if (err)
+        {
+            ERROR("%s: sensors_get_value failed: %s", function_name,
+                  sensors_strerror(err));
+            sensors_free_chip_name(&match);
+            return 0;
+        }
+        snprintf(value, RCF_MAX_VAL, "%f", subfeature_value);
+    }
+    else
+    {
+        snprintf(value, RCF_MAX_VAL, "%s", "");
+    }
+
+    sensors_free_chip_name(&match);
+    return 0;
+}
+
+
+static te_errno
+sensor_min_get(unsigned int gid, const char *oid, char *value,
+               const char *unused1, const char *sensor_str,
+               const char *data_id_str)
+{
+    double val;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(unused1);
+
+    if (get_value_by_name(min_values, sensor_str, data_id_str, &val) == 0)
+        snprintf(value, RCF_MAX_VAL, "%f", val);
+    else
+        snprintf(value, RCF_MAX_VAL, "%s", "");
+
+    return 0;
+}
+
+static te_errno
+sensor_max_get(unsigned int gid, const char *oid, char *value,
+               const char *unused1, const char *sensor_str,
+               const char *data_id_str)
+{
+    double val;
+
+    UNUSED(gid);
+    UNUSED(oid);
+    UNUSED(unused1);
+
+    if (get_value_by_name(max_values, sensor_str, data_id_str, &val) == 0)
+        snprintf(value, RCF_MAX_VAL, "%f", val);
+    else
+        snprintf(value, RCF_MAX_VAL, "%s", "");
+
+    return 0;
+}
+
+static int
+get_value_by_name(struct sensor_value_record **records,
+                  const char *chip_name, const char *feature_name,
+                  double *value)
+{
+    int i;
+
+    for (i = 0; records[i] != NULL; i++)
+    {
+        if (strcmp(records[i]->chip_name, chip_name) == 0 &&
+            strcmp(records[i]->feature_name, feature_name) == 0)
+        {
+            *value = records[i]->value;
+            return 0;
+        }
+    }
+    return -1;
+}
+
+static void
+update_value_by_name(struct sensor_value_record **records,
+                     const char *chip_name, const char *feature_name,
+                     double new_value)
+{
+    int i;
+
+    struct sensor_value_record *new_record;
+
+    for (i = 0; records[i] != NULL; i++)
+    {
+        if (strcmp(records[i]->chip_name, chip_name) == 0 &&
+            strcmp(records[i]->feature_name, feature_name) == 0)
+        {
+            /* Update existing record */
+            records[i]->value = new_value;
+            return;
+        }
+    }
+
+    if (i == SENSOR_ARRAY_MAX_SIZE - 1)
+    {
+        ERROR("Too many sensors for updating minimum/maximum values");
+        return;
+    }
+
+    /* Append new record */
+    new_record = malloc(sizeof(struct sensor_value_record));
+    if (new_record == NULL)
+    {
+        ERROR("update_value_by_name: virtual memory exhausted");
```

Comment:
`TE_FATAL_ERROR` or just leave this unchecked - `TE_ALLOC` is going to do this soon itself